src/main/java/gift/WishListService.java

```diff
@@ -0,0 +1,44 @@
+package gift;
+
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+import java.util.Optional;
+
+@Service
+public class WishListService {
+    private final WishListRepository wishListRepository;
+    private final MemberRepository memberRepository;
+
+    public WishListService(WishListRepository wishListRepository, MemberRepository memberRepository) {
+        this.wishListRepository = wishListRepository;
+        this.memberRepository = memberRepository;
+    }
+
+    public List<WishList> getWishList(String email) {
+        Optional<Member> memberOptional = memberRepository.findMemberByEmail(email);
+        if (memberOptional.isEmpty()) {
+            throw new IllegalArgumentException("존재하지 않는 이메일입니다.");
+        }
+        Member member = memberOptional.get();
+        return wishListRepository.findByMemberId(member.getId());
```

Comment:
```suggestion
        Member member = memberRepository.findMemberByEmail(email)
            .orElseThrow(() -> new IllegalArgumentException("존재하지 않는 이메일입니다."));
        return wishListRepository.findByMemberId(member.getId());
```

Optional 은 이렇게도 활용해볼 수 있어요!
위의 코드를 복붙하지 마시고 직접 입력해서 작성해보는 것을 추천드려요 :)
Optional 이 사용되고 있는 프로젝트의 다른 부분들도 변경해보는 걸 도전해볼까요?